### PR TITLE
Adding annotations to all resources

### LINKED ISSUE #174: Metadata annotations are not propagated to created resources
### What did you do?

Deployed an appsody project with the operator (`appsody deploy`)

### What did you expect to see?

`kubectl describe pod my-project` properly shows the labels from the AppsodyApplication, but it does not show the annotations.

### What did you see instead?

I want to see all of the annotations from AppsodyApplication on all of the resources created by the operator.

### Environment

* Appsody CLI version: 0.4.8

* Appsody Stack: nodejs-express

* Kubernetes version information:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.4", GitCommit:"f49fa022dbe63faafd0da106ef7e05a29721d3f1", GitTreeState:"clean", BuildDate:"2018-12-14T07:10:00Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:41:55Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by JusteenR**
**What this PR does / why we need it?**:

- Adds annotations to resources except for monitoring
- The kAppNav dashboard will be looking for annotations in the next Kabanaro release

**Does this PR introduce a user-facing change?**

- [x] User guide
- [ ] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes #174

### CODE REVIEW

**Comment by leochr** [comment]
@arturdzm Please review. appsodyapplication_types.go  and types.go were changed. Should the auto generated files be updated as well?

**Comment by arturdzm** [comment]
No new fields are added so no regeneration necessary. Also it’s probably good to have GetAnnotations() but not currently required, that’s only useful if plan to add our own,  like for labels

**Comment by navidsh** [comment]
The side effect of this change is that users cannot manually set annotations on resources created by the operator. Annotations can be used for fine-tuning [Route](https://docs.openshift.com/container-platform/3.7/architecture/networking/routes.html#route-specific-annotations) or [Knative Service](https://docs.openshift.com/container-platform/4.2/serverless/configuring-knative-serving-autoscaling.html#knative-serving-concurrent-autoscaling-requests_configuring-knative-serving-autoscaling).

It might be better we merge `AppsodyApplication`'s annotations with resources' annotations.

File: pkg/apis/appsody/v1beta1/appsodyapplication_types.go [hunk 1]
```diff
@@ -514,6 +514,32 @@ func (cr *AppsodyApplication) applyConstants(defaults AppsodyApplicationSpec, co
...
+	}
+
+	return annotations
+}
+
+// GetServiceAnnotations returns a set of annotations to be added to the service
+func (cr *AppsodyApplication) GetServiceAnnotations() map[string]string {
+	annotations := map[string]string{}
+
+	for key, value := range cr.Spec.Service.Annotations {
```
Comment (leochr) [inline pkg/apis/appsody/v1beta1/appsodyapplication_types.go]: @JusteenR Please reverse the order here. Service annotations should be able to override general annotations. Thanks.

Comment (JusteenR) [reply pkg/apis/appsody/v1beta1/appsodyapplication_types.go]: @leochr Service annotations now take precedence over the general ones.

File: pkg/common/types.go [hunk 2]
```diff
@@ -95,4 +95,6 @@ type BaseApplication interface {
 	GetMonitoring() BaseApplicationMonitoring
 	GetLabels() map[string]string
 	GetStatus() BaseApplicationStatus
+	GetAnnotations() map[string]string
+	GetServiceAnnotations() map[string]string
```
[the remark this answers is not in the document]
Comment (navidsh) [reply pkg/common/types.go]: Fixed

**Pull request merged**